internal/pkg/manifest/workload.go

```diff
@@ -66,6 +74,40 @@ type Image struct {
 	DependsOn    map[string]string `yaml:"depends_on,flow"` // Add any sidecar dependencies.
 }
 
+type imageTransformer struct{}
+
+// Transformer implements customized merge logic for Image field of manifest.
+// It merges `DockerLabels` and `DependsOn` in the default manager (i.e. with configurations mergo.WithOverride, mergo.WithOverwriteWithEmptyValue)
+// And then overrides both `Build` and `Location` fields at the same time with the src values, given that they are non-empty themselves.
+func (t imageTransformer) Transformer(typ reflect.Type) func(dst, src reflect.Value) error {
+	if typ != reflect.TypeOf(Image{}) {
+		return nil
+	}
+	return func(dst, src reflect.Value) error {
+		// Perform default merge
+		dstImage := dst.Interface().(Image)
+		srcImage := src.Interface().(Image)
+
+		err := mergo.Merge(&dstImage, srcImage, mergo.WithOverride, mergo.WithOverwriteWithEmptyValue)
+		if err != nil {
+			return err
+		}
+
+		// Perform customized merge
+		dstBuild := dst.Field(0)
+		dstLocation := dst.Field(1)
+
+		srcBuild := src.Field(0)
+		srcLocation := src.Field(1)
+
+		if !srcBuild.IsZero() || !srcLocation.IsZero() {
+			dstBuild.Set(srcBuild)
+			dstLocation.Set(srcLocation)
```

Comment:
Do we need to do any sort of validation here? where users can not set both build and location?

Reply:
The validation will happen afterwards. Specifically, `svc deploy` will read in the manifest, `ApplyEnv` (during which the custom transform happens), and then check if `DockerBuild` is required. During the call to check if a build is required, it will error out if `build` and `location` are set at the same time.

I guess that'd be a better place for that validation than in the transformer. The transformer should probably solely focus on custom merging strategy.